### PR TITLE
Optimize stackgres connection pooling

### LINKED ISSUE #9196: Optimize stackgres connection pooling
### Problem

Further optimize stackgres pgbouncer connection pooling:

- Separate coordinator and worker connection pooling config since each worker node needs to handle way more connections than the coordinator for multi-shard queries
- Increase pgbouncer `max_client_conn` to approximately 2x of `default_pool_size` to allow more client connections to resuse server connections,  also avoid hitting `no more connection available due to max_client_conn` unnecessarily

### Solution

Change the configs and test the performance

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by xin-hedera**
**Description**:

This PR optimizes stackgres connection pooling configs:

- Increase `max_client_connections` and `max_prepared_statements`
- Separate coordinator and worker pgbouncer config
- Set customized stackgres-operator image as the default
- Fix small issues in k6 test contractsResults.js

**Related issue(s)**:

Fixes #9196

**Notes for reviewer**:

js REST api k6 tests show better stability, though it's marginal difference of pass rate, e.g., for `/api/v1/transactions?limit=100`, pass rate increased to 99% or just 32 failed requests, v.s., 98.2% or 6000 failed requests.

Web3 show more improvements, `contractCallBalance` has a pass rate of 98% at 7277/s v.s. in 0.112.0-rc2 99.79% pass rate and 4505.53, in 0.111.0-rc2 99.94% pass rate and 6013.31/s

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: charts/hedera-mirror/values.yaml [hunk 1]
```diff
@@ -313,20 +326,6 @@ stackgres:
     - name: pg_trgm
       version: "1.6"
   nameOverride: citus
-  pgbouncer:
-    pgbouncer:
-      default_pool_size: "800"
-      ignore_startup_parameters: extra_float_digits,options,statement_timeout
-      max_client_conn: "800"
-      max_db_connections: "0"
```
Comment (xin-hedera) [inline charts/hedera-mirror/values.yaml:321]: 0 is the default for `max_db_connections` and `max_user_connections`, so remove them

File: charts/hedera-mirror/values.yaml [hunk 2]
```diff
@@ -296,6 +297,18 @@ stackgres:
     enableMetricsExporter: false
     enablePostgresUtil: true
     instances: 1
+    pgbouncer:
+      pgbouncer:
+        default_pool_size: "800"
+        ignore_startup_parameters: extra_float_digits,options,statement_timeout
+        max_client_conn: "1600"
+        max_prepared_statements: "512"
```
Comment (xin-hedera) [inline charts/hedera-mirror/values.yaml:305]: Increase max prepared statement, pgbouncer doc mentions each prepared statement has a 5KB memory footprint.

No error during testing was observed about improper max prepared statement, just think it's better to start with a large enough value.

File: hedera-mirror-test/k6/src/rest/test/contractsResult.js [hunk 3]
```diff
@@ -30,8 +30,7 @@ const {options, run, setup} = new RestTestScenarioBuilder()
     const url = `${testParameters['BASE_URL_PREFIX']}${getUrl(testParameters)}`;
     return http.get(url);
   })
-  .requiredParameters('DEFAULT_CONTRACT_ID')
-  .check('Contracts id state OK', (r) => isValidListResponse(r, resultListName))
+  .check('Contracts results OK', (r) => isValidListResponse(r, resultListName))
```
Comment (xin-hedera) [inline hedera-mirror-test/k6/src/rest/test/contractsResult.js:33]: the test doesn't require `DEFAULT_CONTRACT_ID`

**Review by jascks — APPROVED** [review]
LGTM

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by edwin-greene — APPROVED** [review]
Looks good

**Pull request merged**